### ISSUE TITLE
buildFHSUserEnv & gcc

### ISSUE THREAD

**Issue by tokudan** [NixOS/nixpkgs#13933]
## Basic info

To make sure that we are on the same page:

* Kernel: (run `uname -a`)

Linux yuna 4.3.4 #1-NixOS SMP Thu Jan 1 00:00:01 UTC 1970 x86_64 GNU/Linux
* System: (NixOS: `nixos-version`, Ubuntu/Fedora: `lsb_release -a`, ...)

15.09.1076.9220f03 (Dingo)
* Nix version: (run `nix-env --version`)

nix-env (Nix) 1.10
* Nixpkgs version: (run `nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion`)

"15.09.1076.9220f03"

## Describe your issue here
I'm using nix-shell to enter a buildFHSUserEnv configured for building openwrt. The targetPkgs contains ncurses, zlib and other derivations required by the build process.
The build fails due to missing headers and libraries.

### Expected result
openwrt is compiled.

### Actual result
error from gcc that ncurses.h and zlib.h cannot be found

### Steps to reproduce
Simplified using only ncurses:
1. See https://gist.github.com/tokudan/93b43e98a65c950c6fcf (pass to nix-shell)
2. Execute: `echo 'int main(int argc, char **argv) { initscr(); return 0; }' | gcc -include ncurses.h -x c -o a.out - -lncurses`
3. When the environment variables in the profile are not set the build fails.

### Possible workaround or solution
In #nixos I got quick help. Setting the environment variables
```
C_INCLUDE_PATH=/usr/include
LIBRARY_PATH=/lib
```
solves the issue for the test case, however these variables are overwritten or unset in the openwrt build process, so a suggestion on IRC was to use the variables
```
C_INCLUDE_PATH=/usr/include
LIBRARY_PATH=/lib
NIX_CFLAGS_COMPILE="-I/usr/include"
NIX_CFLAGS_LINK="-L/usr/lib64 -L/usr/lib32"
```
This solves the issue, openwrt can be built successfully when these variables are set.

On IRC the question was raised if gcc and/or buildFHSUserEnv needs to be updated, this issue is intended as a documentation and follow-up on this.

**Comment by tokudan** [comment]
fixed by 11b6924, i guess.

**Comment by abbradar** [comment]
Thanks, I though it would be auto-closed by Github.
